e2e/test/provisioning/ProvisioningE2ETests.cs

```diff
@@ -530,14 +533,23 @@ public async Task ProvisioningDeviceClient_ValidRegistrationId_MqttWsWithProxy_S
 
             ValidateDeviceRegistrationResult(false, result);
 
+#pragma warning disable CA2000 // Dispose objects before losing scope
+            // The certificate instance referenced in the DeviceAuthenticationWithX509Certificate instance is common for all tests in this class. It is disposed during class cleanup.
             Client.IAuthenticationMethod auth = CreateAuthenticationMethodFromSecurityProvider(security, result.DeviceId);
+#pragma warning restore CA2000 // Dispose objects before losing scope
 
             await ConfirmRegisteredDeviceWorksAsync(result, auth, transportType, false).ConfigureAwait(false);
             await ConfirmExpectedDeviceCapabilitiesAsync(result, auth, deviceCapabilities).ConfigureAwait(false);
 
             if (attestationType != AttestationMechanismType.X509) //x509 enrollments are hardcoded, should never be deleted
             {
-                await DeleteCreatedEnrollmentAsync(enrollmentType, CreateProvisioningService(proxyServerAddress), security, groupId).ConfigureAwait(false);
+                using ProvisioningServiceClient dpsServiceClient = CreateProvisioningService(proxyServerAddress);
+                await DeleteCreatedEnrollmentAsync(enrollmentType, dpsServiceClient, security, groupId).ConfigureAwait(false);
+            }
+
+            if (auth is IDisposable disposableAuth)
```

Comment:
I prefer this approach over the one several files back that used #if !NET451 around it.

Reply:
I like this approach too, however disposing X509Certificates gets a bit messy since NET451 doesn't support that.

Reply:
either it is IDisposable or it isn't, right?

Reply:
For an IAuthneticationMethod yes, it is or isn't IDisposable. 
Did I leave an IAuthenticationMethod dispose with #ifNET451 directive? If yes, then yeah, that isn't necessary. Do you happen to recall where that was, I couldn't find it within the PR.

Reply:
It was some other disposable type. Let me find it.

Reply:
I think I see what you mean, the X509Certificate wouldn't be disposable to begin with in NET451. Yeah, good point.